tests/stress/stress_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package rpc
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/rpc/modules"
+	"github.com/ChainSafe/gossamer/tests/rpc"
+	"github.com/stretchr/testify/require"
+
+	scribble "github.com/nanobox-io/golang-scribble"
+)
+
+var (
+	pidList = make([]*exec.Cmd, 3)
+)
+
+func TestMain(m *testing.M) {
+	if rpc.GOSSAMER_INTEGRATION_TEST_MODE != "stress" {
+		_, _ = fmt.Fprintln(os.Stdout, "Going to skip stress test")
+		return
+	}
+
+	_, _ = fmt.Fprintln(os.Stdout, "Going to start stress test")
+
+	if rpc.NETWORK_SIZE_STR != "" {
+		currentNetworkSize, err := strconv.Atoi(rpc.NETWORK_SIZE_STR)
+		if err == nil {
+			_, _ = fmt.Fprintln(os.Stdout, "Going to custom network size ... ", "currentNetworkSize", currentNetworkSize)
+			pidList = make([]*exec.Cmd, currentNetworkSize)
+		}
+	}
+
+	// Start all tests
+	code := m.Run()
+	os.Exit(code)
+}
+
+func TestStressSync(t *testing.T) {
+	t.Log("going to start TestStressSync")
+	localPidList, err := rpc.Bootstrap(t, pidList)
+	require.Nil(t, err)
+
+	tempDir, err := ioutil.TempDir("", "gossamer-stress-db")
+	require.Nil(t, err)
+	t.Log("going to start a JSON simple database to track all chains")
+
+	db, err := scribble.New(tempDir, nil)
+	require.Nil(t, err)
+
+	blockHighestBlockHash := "chain_getHeader"
+
+	for i, v := range localPidList {
+
+		t.Log("going to get HighestBlockHash from node", "i", i, "v", v)
+
+		//Get HighestBlockHash
+		respBody := rpc.PostRPC(t, blockHighestBlockHash, "http://"+rpc.GOSSAMER_NODE_HOST+":854"+strconv.Itoa(i))
+
+		// decode resp
+		target := rpc.DecodeRPC(t, respBody, blockHighestBlockHash)
+
+		// convert
+		chainBlockResponse, ok := target.(modules.ChainBlockHeaderResponse)
+		require.True(t, ok)
+
+		err = db.Write("blocks_"+strconv.Itoa(v.Process.Pid),
```

Comment:
why are we writing to the db here?

Reply:
and then use the json output in a nice UI where we can play with the generated data

Reply:
I don't think we should be storing the results in the database... if you are storing the results from the test, would be nice to output to a test log and/or within a git ignored directory rather than using the database module to store test data but maybe I am misunderstanding...?

Reply:
We are storing with a db package that is actually just a JSON file. It is being stored in a temp folder, so I can on the next test start playing with queries on it and further expand the tests.